common-npm-packages/azurermdeploycommon/webdeployment-common/ziputility.ts

```diff
@@ -2,28 +2,23 @@ import tl = require('azure-pipelines-task-lib/task');
 import path = require('path');
 import Q = require('q');
 import fs = require('fs');
-import StreamZip = require('node-stream-zip');
-var DecompressZip = require('decompress-zip');
+import StreamZip from 'node-stream-zip';
 var archiver = require('archiver');
 
 export async function unzip(zipLocation, unzipLocation) {
     var defer = Q.defer();
     if(tl.exist(unzipLocation)) {
       tl.rmRF(unzipLocation);
     }
-    var unzipper = new DecompressZip(zipLocation);
-    tl.debug('extracting ' + zipLocation + ' to ' + unzipLocation);
-    unzipper.on('error', function (error) {
-        defer.reject(error);
-    });
-    unzipper.on('extract', function (log) {
-        tl.debug('extracted ' + zipLocation + ' to ' + unzipLocation + ' Successfully');
-        defer.resolve(unzipLocation);
-    });
-    unzipper.extract({
-      path: unzipLocation
-    });
-    return defer.promise;
+    tl.mkdirP(unzipLocation);
+    const zip = new StreamZip.async({ file: zipLocation});
```

Comment:
[Not blocking as I'm sure it's well-tested]:  but it would be nice if the new code path were switched using a parameter so we can fall back to the old code path for node16.  E.g. if node20 + ode-stream-zip has a regression, we can use the buildconfig to revert to old code path.

Reply:
@merlynomsft we can downgrade version of this package in case if we will face problem, and rollback changes in the next version of common package